hw/bsp/frdm-k64/src/os_bsp.c

```diff
@@ -48,32 +48,34 @@ void _close(int fd);
 
 /*
  * XXXX for now have it here.
+ * Due to flash driver issue in Zephyr
+ * Shrink usable flash area to 512k
  */
 static struct flash_area bsp_flash_areas[] = {
     [FLASH_AREA_BOOTLOADER] = {
         .fa_flash_id = 0,       /* internal flash */
-        .fa_off = 0x00000000,   /* beginning */
-        .fa_size = (40 * 1024)
+        .fa_off = 0x00008000,   /* beginning */
+        .fa_size = (32 * 1024)
     },
     [FLASH_AREA_IMAGE_0] = {
         .fa_flash_id = 0,
-        .fa_off = 0x0000A000,
-        .fa_size = (484 * 1024)
+        .fa_off = 0x00020000,
+        .fa_size = (128 * 1024)
     },
     [FLASH_AREA_IMAGE_1] = {
         .fa_flash_id = 0,
-        .fa_off = 0x00083000,
-        .fa_size = (484 * 1024)
+        .fa_off = 0x00040000,
+        .fa_size = (128 * 1024)
     },
     [FLASH_AREA_IMAGE_SCRATCH] = {
         .fa_flash_id = 0,
-        .fa_off = 0x000fc000,
-        .fa_size = (8 * 1024)
+        .fa_off = 0x00060000,
+        .fa_size = (128 * 1024)
```

Comment:
Scratch doesn't need to be that big, does it? The idea is to have it large enough that can store an entire sector, so based on the previous change, 8 should be enough (not sure about the sector size for this board).

Then we can at least increase the image 0 and 1, as while 128 is enough, it would be good to have a bit more.